app/src/main/java/topads/tokopedia/com/tinkersample/UserPrefs.kt

```diff
@@ -0,0 +1,32 @@
+package topads.tokopedia.com.tinkersample
+
+import android.content.Context
+import android.content.SharedPreferences
+
+/**
+ * Author errysuprayogi on 10,January,2020
+ */
+class UserPrefs private constructor(private val context: Context) {
+    private val settings: SharedPreferences
+    fun saveString(key: String?, value: String?) {
+        settings.edit().putString(key, value).commit()
+    }
+
+    fun getString(key: String?, defaultValue: String?): String? {
+        return settings.getString(key, defaultValue)
+    }
+
+    companion object {
+        private var instance: UserPrefs? = null
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/errysuprayogi/TinkerSampleApp/blob/eb257e9dc5962f6a2ae2e1d0dddb2d5a7ded5da8/app/src/main/java/topads/tokopedia/com/tinkersample/UserPrefs.kt#L20"></span>Do not place Android context classes in static fields (static reference to <code>UserPrefs</code> which has field <code>context</code> pointing to <code>Context</code>); this is a memory leak</td>
    </tr>
  </tbody>
</table>